README.dev.md

```diff
@@ -6,11 +6,612 @@ You will need a machine with at least 2GB of RAM available (see
+* [Using Docker (experimental)](#using-docker-experimental)
+
+## First-time contributors start here
+
+This section guides first-time contributors through installing the Zulip dev
+environment on Windows 10, OS X El Capitan, Ubuntu 14.04, and Ubuntu 16.04. 
+
+Those who have installed Zulip before or are experienced at using Vagrant may
+wish to skip ahead to [Using Vagrant](#using-vagrant), [Using
+Docker](#using-docker-experimental), or read about installing [By
+hand](#by-hand).
+
+The recommended method for installing the Zulip dev environment is to use
+Vagrant with VirtualBox on Windows and OS X, and Vagrant with LXC on
+Ubuntu. This method creates a virtual machine (for Windows and OS X) 
+or a Linux container (for Ubuntu) inside which the Zulip server and
+all related services will run. 
+
+Contents:
+* [Requirements](#requirements)
+* [Step 1: Install Prerequisites](#step-1-install-prerequisites)
+* [Step 2: Get Zulip code](#step-2-get-zulip-code)
+* [Step 3: Start the dev environment](#step-3-start-the-dev-environment)
+* [Step 4: Developing](#step-4-developing)
+* [Troubleshooting & Common Errors](#troubleshooting--common-errors)
+
+If you encounter errors installing the Zulip dev environment and they are not addressed in [Troubleshooting & Common Errors](#troubleshooting--common-errors), send a note to the [Zulip-devel Google group](https://groups.google.com/forum/#!forum/zulip-devel) or [file an issue](https://github.com/zulip/zulip/issues).
+
+### Requirements
+
+Installing the Zulip dev environment requires downloading dependencies of
+several hundred megabytes. You will need an active internet connection
+throughout the entire installation processes.
+
+- **Windows**: Windows 10 64-bit, hardware virtualization enabled (VT-X or AMD-V), administrator access, [Cygwin][cygwin-dl], [VirtualBox][vbox-dl], [Vagrant][vagrant-dl].
+- **OS X**: OS X El Capitan, Git, [VirtualBox][vbox-dl], [Vagrant][vagrant-dl].
+- **Ubuntu**: 14.04 64-bit or 16.04 64-bit, Git, [Vagrant][vagrant-dl], lxc.
+- **All**: 2GB available RAM, Active broadband internet connection.
+
+[cygwin-dl]: http://cygwin.com/
+
+### Step 1: Install Prerequisites
+
+#### Windows 10
+
+1. Install [Cygwin][cygwin-dl]. Make sure to install default required
+   packages along with **git**, **curl**, **openssh**, and **rsync**
+   binaries.
+2. Install [VirtualBox][vbox-dl]
+3. Install [Vagrant][vagrant-dl]
+
+Now you are ready for [Step 2: Get Zulip Code.](#step-2-get-zulip-code)
+
+#### OS X
+
+1. Install [VirtualBox][vbox-dl]
+2. Install [Vagrant][vagrant-dl]
+
+Now you are ready for [Step 2: Get Zulip Code.](#step-2-get-zulip-code)
+
+#### Ubuntu 14.04
+
+##### 1. Install Vagrant
+
+For 14.04 Trusty you'll need a more recent version of Vagrant than what's
+available in the official Ubuntu repositories. 
+
+First uninstall any vagrant package you may have installed from the Ubuntu
+repository:
+
+```
+christie@trusty-desktop:~
+$ sudo apt-get purge vagrant
```

Comment:
So I like that this style makes it clear which machine you're running these things on, but it is somewhat verbose (and I guess regardless it's hard to just copy-paste since we have the `$`s at the start of the lines) and for users who just want to get installed quickly (as opposed to understand what's happening), it's not ideal.

I wonder if there's a good way to also provide the "just copy-paste this 3-4 command block" version, at least for Linux where that's a coherent thing to do (doesn't really make sense on windows/osx).